beacon-chain/p2p/peers/scorers/data_column_rpc_request_scorer.go

```diff
@@ -0,0 +1,153 @@
+	DefaultDataColumnRPCRequestPenaltyFactor = float64(0.02)
+)
+
+// DataColumnRPCRequestScorer represents scoring service for tracking data column RPC requests.
+type DataColumnRPCRequestScorer struct {
+	config *DataColumnRPCRequestScorerConfig
+	store  *peerdata.Store
+}
+
+// DataColumnRPCRequestScorerConfig holds configuration parameters for data column RPC request scoring service.
+type DataColumnRPCRequestScorerConfig struct {
+	// DecayInterval defines how often stats should be decayed.
+	DecayInterval time.Duration
+	// Decay specifies number of requests subtracted from stats on each decay step.
+	Decay uint64
+	// Threshold defines maximum number of requests before peer is considered bad.
+	Threshold uint64
+	// PenaltyFactor defines multiplier applied to request count when calculating score.
+	PenaltyFactor float64
+}
+
+// newDataColumnRPCRequestScorer creates new scoring service for data column RPC requests.
+func newDataColumnRPCRequestScorer(store *peerdata.Store, config *DataColumnRPCRequestScorerConfig) *DataColumnRPCRequestScorer {
+	// Create a new config with default values
+	cfg := &DataColumnRPCRequestScorerConfig{
+		DecayInterval: DefaultDataColumnRPCRequestDecayInterval,
+		Decay:         DefaultDataColumnRPCRequestDecay,
+		Threshold:     DefaultDataColumnRPCRequestThreshold,
+		PenaltyFactor: DefaultDataColumnRPCRequestPenaltyFactor,
+	}
+
+	// Override with provided config values if present
+	if config != nil {
+		if config.DecayInterval != 0 {
+			cfg.DecayInterval = config.DecayInterval
+		}
+		if config.Decay != 0 {
+			cfg.Decay = config.Decay
+		}
+		if config.Threshold != 0 {
+			cfg.Threshold = config.Threshold
+		}
+		if config.PenaltyFactor != 0 {
+			cfg.PenaltyFactor = config.PenaltyFactor
+		}
+	}
+
+	return &DataColumnRPCRequestScorer{
+		config: cfg,
+		store:  store,
+	}
+}
+
+// Score returns calculated peer score.
+func (s *DataColumnRPCRequestScorer) Score(pid peer.ID) float64 {
+	s.store.RLock()
+	defer s.store.RUnlock()
+	return s.scoreNoLock(pid)
+}
+
+// scoreNoLock is a lock-free version of Score.
+func (s *DataColumnRPCRequestScorer) scoreNoLock(pid peer.ID) float64 {
+	if peerData, ok := s.store.PeerData(pid); ok {
+		// Apply penalty based on request count
+		score := -1 * float64(peerData.DataColumnRequestCount) * s.config.PenaltyFactor
+		return math.Round(score*ScoreRoundingFactor) / ScoreRoundingFactor
+	}
+	return 0
+}
+
+// IsBadPeer implements Scorer interface.
+func (s *DataColumnRPCRequestScorer) IsBadPeer(pid peer.ID) error {
+	s.store.RLock()
+	defer s.store.RUnlock()
+	return s.isBadPeerNoLock(pid)
+}
+
+// isBadPeerNoLock is a lock-free version of IsBadPeer.
+func (s *DataColumnRPCRequestScorer) isBadPeerNoLock(pid peer.ID) error {
+	if peerData, ok := s.store.PeerData(pid); ok && peerData.DataColumnRequestCount >= s.config.Threshold {
+		return errors.New("exceeded data column request threshold")
```

Comment:
Will be nice to get the numbers: 
```
fmt.Errorf("bad peer %s: request count %d exceeds threshold %d", pid, peerData.DataColumnRequestCount, s.config.Threshold)
```

Reply:
[f29362e](https://github.com/OffchainLabs/prysm/pull/15205/commits/f29362eb6867b518efa8f38f101dc63cb9bad64a)